scalardl/src/jepsen/scalardl.clj

```diff
@@ -0,0 +1,159 @@
+                    value (s/response->int (.executeContract conn "read" argument))]
+                (assoc op :type :ok, :value (independent/tuple k value)))
+        :write (let [argument (s/create-argument k v)]
+                 (.executeContract conn "write" argument)
+                 (assoc op :type :ok)))))
+
+  (teardown! [_ test]))
+
+(defn db
+  [version]
+  (reify db/DB
+    (setup! [_ test node]
+      (s/spinup-cassandra! test node)
+
+      (c/upload [(str ledger-dir "/ledger.tar")
+                 (str ledger-dir "/ledger.properties")
+                 (str ledger-dir "/client.pem")
+                 (str ledger-dir "/client-key.pem")]
+                "/tmp")
+
+      (when (= node (first (:nodes test)))
+           (info node "creating schema for scalardl")
+           (s/create-schema! (vector node) (:rf test)))
+
+      (info node "installing scalardl")
+      (cu/install-archive! "file:///tmp/ledger.tar" dir)
+      (c/exec (c/lit (str "sed -i -e s/scalar.database.contact_points=.*/scalar.database.contact_points="
+                          node
+                          "/ /tmp/ledger.properties")))
+
+      (cu/start-daemon!
+        {:logfile logfile
+         :pidfile pidfile
+         :chdir   dir}
+        binary
+        :-config "/tmp/ledger.properties")
+      (Thread/sleep 2000))  ; sleeping is necessary?
+
+    (teardown! [_ test node]
+      (info node "tearing down scalardl")
+      (cu/stop-daemon! binary pidfile)
+      (c/su (c/exec :rm :-rf dir))
+      (s/teardown-cassandra! node))
+
+    db/LogFiles
+    (log-files [_ test node]
+      [logfile
+       "/root/cassandra/logs/system.log"])))
+
+(def cli-opts
+  [[nil "--rf REPLICATION_FACTOR" "Replication factor"
+    :default 3
+    :parse-fn parse-long
+    :validate [pos? "Must be a positive integer."]]
+   ["-r" "--rate HZ" "Approximate number of requests per second, per thread."
+    :default 10
+    :parse-fn read-string
+    :validate [#(and (number? %) (pos? %)) "Must be a positive number."]]
+   [nil "--ops-per-key NUM" "Maximum number of operations on any given key."
+    :default 100
+    :parse-fn parse-long
+    :validate [pos? "Must be a positive integer."]]
+   (cli/tarball-opt "https://archive.apache.org/dist/cassandra/3.11.4/apache-cassandra-3.11.4-bin.tar.gz")])
+
+(defn scalardl-test
+  [opts]
+  (merge tests/noop-test
+         opts
+         {:name           "scalardl"
+          :os             debian/os
+          :db             (db "3.11.4")
+          :decommissioned (atom #{})    ; needed to avoid NullPointerExceptions in Cassandra tests
+          :client         (Client. nil)
+          :nemesis        (nemesis/partition-random-halves)
+          :checker        (independent/checker
+                            (checker/linearizable {:model     (model/register)
+                                                   :algorithm :linear}))
+          :generator      (->> (independent/concurrent-generator
```

Comment:
Just a comment. We will inject various nemeses in another PR.
If you use `conductor/std-gen`, you might be able to inject nemesis easier.

Reply:
Thank you. I will take a look at it.